dozer-api/src/grpc/internal/internal_pipeline_server.rs

```diff
@@ -6,45 +6,71 @@ use crate::grpc::internal_grpc::{
 use crossbeam::channel::Receiver;
 use dozer_types::{crossbeam, log::info, models::app_config::Config, tracing::warn};
 use std::{net::ToSocketAddrs, pin::Pin};
-use tokio::runtime::Runtime;
-use tokio_stream::{wrappers::ReceiverStream, StreamExt};
+use tokio::{
+    runtime::Runtime,
+    sync::broadcast::{self, Sender},
+};
+use tokio_stream::wrappers::ReceiverStream;
 use tonic::{codegen::futures_core::Stream, transport::Server, Response, Status};
 
 pub struct InternalPipelineServer {
     app_config: Config,
-    receiver: Receiver<PipelineResponse>,
+    receiver: broadcast::Receiver<PipelineResponse>,
+}
+impl InternalPipelineServer {
+    pub fn new(app_config: Config, receiver: Receiver<PipelineResponse>) -> Self {
+        let (tx, rx1) = broadcast::channel::<PipelineResponse>(16);
+        tokio::spawn(async move {
+            Self::setup_broad_cast_channel(tx, receiver);
+        });
+        Self {
+            app_config,
+            receiver: rx1,
+        }
+    }
+
+    fn setup_broad_cast_channel(
+        tx: Sender<PipelineResponse>,
+        receiver: Receiver<PipelineResponse>,
+    ) {
+        while let Ok(event_response) = receiver.recv() {
+            let _ = tx.send(event_response);
```

Comment:
Maybe print a log if it's an `Err`?

Reply:
yah, i want to get rid of `crossbeam` also,  but it using across everywhere in `dozer-orchestrator` gonna have a huge affect on that.

`broadcast` is for solving issue multiple clients subscribe to that method. Just found out that bug: when a client is subscribe, later client subscribe won't receive any message